Intermittent test failure on Wayland testbed tests

### Describe the bug

There is an intermittent error on the linux-wayland testbed backend test. 

### Steps to reproduce

Difficult to reproduce; it occurs in CI on the linux-wayland testbed target:
```
=================================== FAILURES ===================================
____________________________ test_save_as_document _____________________________
Traceback (most recent call last):
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/runner.py", line 341, in from_call
    result: TResult | None = func()
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/runner.py", line 242, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/threadexception.py", line 92, in pytest_runtest_call
    yield from thread_exception_runtest_hook()
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/threadexception.py", line 68, in thread_exception_runtest_hook
    yield
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/unraisableexception.py", line 95, in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/unraisableexception.py", line 70, in unraisable_exception_runtest_hook
    yield
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/logging.py", line 848, in pytest_runtest_call
    yield from self._runtest_for(item, "call")
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/logging.py", line 831, in _runtest_for
    yield
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/capture.py", line 879, in pytest_runtest_call
    return (yield)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 122, in _multicall
    teardown.throw(exception)  # type: ignore[union-attr]
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/skipping.py", line 257, in pytest_runtest_call
    return (yield)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/runner.py", line 174, in pytest_runtest_call
    item.runtest()
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pytest_asyncio/plugin.py", line 457, in runtest
    super().runtest()
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 182, in _multicall
    return outcome.get_result()
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_result.py", line 100, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/_pytest/python.py", line 159, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app_packages/pytest_asyncio/plugin.py", line 929, in inner
    _loop.run_until_complete(task)
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app/tests/conftest.py", line 133, in run_until_complete
    return asyncio.run_coroutine_threadsafe(coro, self.loop).result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/runner/work/toga/toga/testbed/build/testbed/ubuntu/jammy/testbed-0.0.1/usr/lib/testbed/app/tests/app/test_document_app.py", line 154, in test_save_as_document
    app.documents[0]._content.write.assert_called_with(
  File "/usr/lib/python3.10/unittest/mock.py", line 920, in assert_called_with
    raise AssertionError(error_message)
AssertionError: expected call not found.
Expected: write(PosixPath('/tmp/pytest-of-runner/pytest-0/test_save_as_document0/new_filename.testbed'))
Actual: not called.
=========================== short test summary info ============================
FAILED tests/app/test_document_app.py::test_save_as_document - AssertionError: expected call not found.
Expected: write(PosixPath('/tmp/pytest-of-runner/pytest-0/test_save_as_document0/new_filename.testbed'))
Actual: not called.
======= 1 failed, 444 passed, 59 skipped, 7 xfailed in 105.96s (0:01:45) =======

[testbed] Test suite failed!
```

### Expected behavior

The test should pass. It usually *does* pass if you re-run the Wayland test.

### Screenshots

_No response_

### Environment

- Operating System: CI
- Python version: 3.10
- Software versions:
  - Briefcase: 0.3.19+
  - Toga: 0.4.6+
  - ...


### Logs

```

```


### Additional context

The error is presumably timing related; something is happening too quickly (or too slowly) under CI conditions. However, I've been unable to replicate the error locally.

In the case of `test_save_all_documents`, it calls `app.documents.save_all()` which iterates all windows and calls `save()` for each one; so, the test should be insensitive to which window has focus.

As for `test_save_document`, it seems this may be an order of events thing; I was able to create the issue for `test_save_document` by running `test_save_as_document` first.
```
AssertionError: expected call not found.
Expected: write(PosixPath('/home/russell/github/beeware/toga/testbed/build/testbed/pop/jammy/testbed-0.0.1/usr/lib/testbed/app/tests/app/docs/example.testbed'))
Actual: not called.
=========================== short test summary info ============================
FAILED tests/app/test_document_app.py::test_save_document - AssertionError: expected call not found.
```

--- a/changes/2807.misc.rst
+++ b/changes/2807.misc.rst
@@ -1,0 +1,1 @@
+The tests for Testbed no longer error for Document save as testing.

--- a/testbed/tests/app/test_document_app.py
+++ b/testbed/tests/app/test_document_app.py
@@ -26,7 +26,7 @@ async def test_new_document(app, app_probe):
 
     # Document has not been read
     app.documents[0]._content.read.assert_not_called()
-    app.documents[0].title == "Example document: Untitled"
+    assert app.documents[0].title == "Example Document: Untitled"
 
 
 async def test_open_document(app, app_probe):
@@ -35,7 +35,7 @@ async def test_open_document(app, app_probe):
     document_path = Path(__file__).parent / "docs/example.testbed"
     app.documents.open(document_path)
 
-    await app_probe.redraw("Document has been opened")
+    await app_probe.redraw("Document has been opened", delay=0.1)
 
     assert len(app.documents) == 1
     assert len(app.windows) == 2
@@ -109,7 +109,7 @@ async def test_save_document(app, app_probe):
     document_path = Path(__file__).parent / "docs/example.testbed"
     app.documents.open(document_path)
 
-    await app_probe.redraw("Document has been opened")
+    await app_probe.redraw("Document has been opened", delay=0.1)
 
     assert len(app.documents) == 1
     assert len(app.windows) == 2
@@ -138,7 +138,7 @@ async def test_save_as_document(monkeypatch, app, app_probe, tmp_path):
 
     monkeypatch.setattr(document.main_window, "dialog", mock_save_as_dialog)
 
-    await app_probe.redraw("Document has been opened")
+    await app_probe.redraw("Document has been opened", delay=0.1)
 
     assert len(app.documents) == 1
     assert len(app.windows) == 2
@@ -162,7 +162,7 @@ async def test_save_all_documents(app, app_probe):
     document_path = Path(__file__).parent / "docs/example.testbed"
     app.documents.open(document_path)
 
-    await app_probe.redraw("Document has been opened")
+    await app_probe.redraw("Document has been opened", delay=0.1)
 
     assert len(app.documents) == 1
     assert len(app.windows) == 2

